Fixed last pr

BasKiers opened the pull request:
Re-added lost changes from sprut
Fixed discrepancies between app.json and the config directory
Fixed merge conflict in app.json
Fixed default value for 'alarm_motion_reset_window' to be 300ms

BasKiers commented:
To fix the merge conflicts which are all in the name translations of drivers the FIRST entry should be kept and the second one deleted along with the ====='s

TedTolboom commented:
@BasKiers Thanks! Majority of PR I'd corrected already together with some cleanup and pushed to the repository...

On the "Fixed default value for 'alarm_motion_reset_window' to be 300ms" I do not agree to the change. IF there is no value for "alarm_motion_reset_window", it should take the 300...
The part where I don't agree is that it is 300ms... this should be the same unit (seconds)...

The motion sensor has a re-trigger interval of 60 seconds... so I think a 300ms cancellation time does not make sense..

BasKiers commented:
Hmm sorry this pull request got mixed up with a new feature I wanted to PR separately since I thought this pr was merged (: If you want I can split this pr up but if this is unnecessary I will leave it like this (:

The new feature I added is one you requested over slack. I added a script.js file to the config folder which exports a function that is called by the node-homey-config-composer just before everything is written to the app.json file. Here I added a function that filters all drivers and compares the "branch" value of the driver with the name of the current branch. This will result in the following behaviour: 
if branch === production drivers with "branch" production will be added to app.json
if branch === beta drivers with "branch" production or beta will be added to app.json
if branch === alpha drivers with "branch" production or beta or alpha will be added to app.json
else all drivers will be added to the app.json.

This makes it possible to quickly merge/build the app.json config from the SDK2 branch to the production branch without manually changing driver files. This does however requires the latest version of node-homey-config-composer so you will need to update the config composer by executing 'npm install -g node-homey-config-composer' in your terminal
